### PR TITLE
add test

### PR DESCRIPTION
**Pull request by cielavenir**
how about adding tests?

(note: additional test is performed by `7z t` if 7z impl supports zstd (i.e. https://github.com/jinfeihan57/p7zip))

partially from https://github.com/cielavenir/zipfile39/blob/master/test/test_zipfile.py

### CODE REVIEW

**Comment by cielavenir** [comment]
the test is done although I need your opinion

unwip is better

File: test/test_zipfile.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import itertools
+import pytest
+try:
+    from tempfile import TemporaryDirectory
+except ImportError:
+    from backports.tempfile import TemporaryDirectory
+try:
+    from inspect import signature
+except ImportError:
+    from funcsigs import signature
```
Comment (Akaricchi) [inline test/test_zipfile.py]: These fallbacks are probably unnecessary, given the Python 3.6+ requirement.

**Review by Akaricchi — APPROVED** [review]

File: test/test_zipfile.py (hunk 1 above)
Comment (cielavenir) [reply test/test_zipfile.py]: done.

excuse: zipfile39 supports python2, but I should remove such polyfill for zipfile-zstd dedicated test.

Comment (Akaricchi) [reply test/test_zipfile.py]: Oh, I think we should at least give credit to the original file in a comment.

Comment (cielavenir) [reply test/test_zipfile.py]: sure 

I have not thought of the credit in depth as it was written by me

maybe I should add the credit for further reference (for those who comes here before visiting zipfile39 repo)

**Pull request merged**